src/http/response.cr

```diff
@@ -70,8 +78,9 @@ class HTTP::Response
       http_version, status_code, status_message = line.split(3)
       status_code = status_code.to_i
       status_message = status_message.chomp
+      mandatory_body = !(status_code / 100 == 1 || status_code == 204 || status_code == 304)
```

Comment:
This faily complex condition is duplicated at two places, I wonder if we can find a place to extract it. Maybe just `HTTP::Response.mandatory_body?(status_code : Int32)`?